apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,242 @@
+
+  const totalTestCount = countTotalTests(groupedTestsMap);
+  const totalFilteredTestCount = countTotalTests(filteredGroupedTestsMap);
+
+  const hasMatchingResults = totalFilteredTestCount > 0;
+  const hasResults = data && totalTestCount > 0;
+  return (
+    <Container>
+      {loading ? (
+        <ListSkeleton />
+      ) : (
+        <>
+          <Title>
+            {numberOfTestsThatFailedOnMoreThanOneTask}{" "}
+            {pluralize("test", numberOfTestsThatFailedOnMoreThanOneTask)} failed
+            across more than one task
+          </Title>
+          <Body>
+            This page provides an overview of all test failures in the current
+            version. If a test fails across multiple tasks, it could suggest a
+            flaky test or a broader issue.
+          </Body>
+          <FilterContainer>
+            <div>
+              <LabelWrapper>
+                <Label
+                  htmlFor="test-failure-search"
+                  id="test-failure-search-label"
+                >
+                  Search Test Failures
+                </Label>
+              </LabelWrapper>
+              <TextInputWithValidation
+                aria-labelledby="test-failure-search-label"
+                disabled={!hasResults}
+                id="test-failure-search-input"
+                onChange={(value) => {
+                  setTestName(value);
+                }}
+                placeholder="Search failed tests (regex)"
+                validator={validateRegexp}
+              />
+            </div>
+            <Combobox
+              disabled={!hasResults}
+              label="Failure type"
+              multiselect
+              onChange={setSelectedTaskStatuses}
+              placeholder="Select a task status"
+              value={selectedTaskStatuses}
+            >
+              {taskStatuses.map((status) => (
+                <ComboboxOption
+                  key={status}
+                  displayName={taskStatusToCopy[status]}
+                  value={status}
+                />
+              ))}
+            </Combobox>
+            <Combobox
+              disabled={!hasResults}
+              label="Build Variant"
+              multiselect
+              onChange={setSelectedBuildVariants}
+              overflow="scroll-x"
+              placeholder="Select a build variant"
+              value={selectedBuildVariants}
+            >
+              {buildVariants.map((bv) => (
+                <ComboboxOption
+                  key={bv.buildVariant}
+                  displayName={bv.buildVariantDisplayName}
+                  value={bv.buildVariant}
+                />
+              ))}
+            </Combobox>
+          </FilterContainer>
+          {hasResults && (
+            <FilterSubheaderContainer>
+              <Body weight="medium">
+                {totalFilteredTestCount}/{totalTestCount} Filtered Failed{" "}
```

Comment:
Would suggest omitting "filtered" because using a fraction implies filtering in a UI imo. It's also slightly confusing if no filters are applied
```suggestion
                {totalFilteredTestCount}/{totalTestCount} Failed{" "}
```